sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -274,12 +274,53 @@ class MapboxMap :
     onMapLoadErrorListener: OnMapLoadErrorListener? = null,
   ) {
     checkNativeMap("loadStyle")
-    this.loadStyleUri(
-      styleExtension.styleUri,
-      transitionOptions,
-      { style -> onFinishLoadingStyleExtension(style, styleExtension, onStyleLoaded) },
-      onMapLoadErrorListener
+    initializeStyleLoad(
+      onStyleLoaded = { style ->
+        // TODO order of styleDataSourcesLoadedListener / styleDataSpritesLoadedListener is not defined
+        // and the callbacks are optional - e.g. style without sources / sprites won't trigger those events.
+        // To make sure every style extension component is loaded
+        // we bind them in the final onStyleLoaded callback.
+        // Need to revisit in 10.6.0-beta.2
+        styleExtension.images.forEach {
+          it.bindTo(style)
+        }
+        styleExtension.sources.forEach {
+          it.bindTo(style)
+        }
+        styleExtension.layers.forEach { (layer, layerPosition) ->
+          layer.bindTo(style, layerPosition)
+        }
+        onStyleLoaded?.onStyleLoaded(style)
+      },
+      styleDataStyleLoadedListener = { style ->
+        styleExtension.light?.bindTo(style)
+        styleExtension.terrain?.bindTo(style)
+        styleExtension.atmosphere?.bindTo(style)
+        styleExtension.projection?.bindTo(style)
+        transitionOptions?.let(style::setStyleTransition)
+      },
+      styleDataSourcesLoadedListener = { style ->
+        // TODO
+//        styleExtension.sources.forEach {
+//          it.bindTo(style)
+//        }
+//        styleExtension.layers.forEach { (layer, layerPosition) ->
+//          layer.bindTo(style, layerPosition)
+//        }
+      },
+      styleDataSpritesLoadedListener = { style ->
+        // TODO
+//        styleExtension.images.forEach {
+//          it.bindTo(style)
+//        }
+      },
```

Comment:
nits: remove?

Reply:
Since we're going to uncomment soon - I'd kept it here